--- python-efl-1.10.1/efl/ecore/efl.ecore_events.pxi.orig	2014-05-20 21:50:41.000000000 +0200
+++ python-efl-1.10.1/efl/ecore/efl.ecore_events.pxi	2015-10-18 16:42:20.015506777 +0200
@@ -301,7 +301,7 @@
 
 
 cdef class CustomEvent(Event):
-    cdef int _set_obj(self, void *obj):
+    cdef int _set_obj(self, void *obj) except 0:
         self.obj = <object>obj
         return 1
 
@@ -328,14 +328,14 @@
         self._set_obj(ecore_event_add(type, <void *>self.args, _event_free_cb,
                       <void*>self))
 
-    cdef int _set_obj(self, Ecore_Event *ev):
+    cdef int _set_obj(self, Ecore_Event *ev) except 0:
         assert self.obj == NULL, "Object must be clean"
         assert ev != NULL, "Cannot set NULL as object"
         self.obj = ev
         Py_INCREF(self)
         return 1
 
-    cdef int _unset_obj(self):
+    cdef int _unset_obj(self) except 0:
         if self.obj != NULL:
             self.obj = NULL
             self.args = None
--- python-efl-1.10.1/efl/ecore/x.pxd.orig	2014-08-18 08:18:36.000000000 +0200
+++ python-efl-1.10.1/efl/ecore/x.pxd	2015-10-18 17:04:57.242145154 +0200
@@ -1310,7 +1310,7 @@
 cdef class Window:
     cdef readonly Ecore_X_Window xid
 
-    cdef int _set_xid(self, Ecore_X_Window xid)
+    cdef int _set_xid(self, Ecore_X_Window xid) except 0
 
 
 cdef class EventKey(Event):
